### DISCUSSION TITLE
Has this project considered a "rewrite in rust"? Seems popular at the moment, and most projects seem happy with their decision, but it's likely a lot of work.

### DISCUSSION THREAD

**Discussion by DeflateAwning** [steveicarus/iverilog#1089] [General]
Just curious and want to start the discussion, as this is a very useful project with tons of users. 

Projects like the `fish` shell have had very successful rewrites in rust which have allowed them to add features which would otherwise be difficult (largely around async and thread support). As performance is a consideration in simulation projects, it seems like it _could_ be appropriate. 

Excited to hear any thoughts!

**Comment by caryr** [comment]
While I have interest in `rust` I have not had time to actually work with it and before I and I believe the rest of the group would consider something like this we would need someone in the primary team who was fluent in `rust` to help guide the rest of the team. We also have very limited time to work on Icarus so a complete rewrite given the size of the development team seems impractical. I believe that for now it is better for us to focus what time we do have on resolving bugs and adding the simpler enhancement requests. There are a couple bigger changes that I have been planned for year, but I never seem to have time to actually work on them.
